File: 48/source.java
import java.math.BigInteger;
public class e48 {

	private static void lastTen(BigInteger a){
		String s = a.toString();
		String reverse = new StringBuffer(s).reverse().toString();
		String result = new StringBuffer(reverse.substring(0, 10)).reverse().toString();
		System.out.println(result);
	}
	
	public static void main(String args[]){
		BigInteger sum = new BigInteger("0");
		BigInteger tmp = new BigInteger("0");
		for(int i = 1; i <= 1000; i++){
			tmp = BigInteger.valueOf(i).pow(i);
			sum = sum.add(tmp);
		}
		lastTen(sum);	
	}
}

File: 63/source.py
from math import pow

def check(n,e):
    tmp = int(pow(n,e))
    return len(str(tmp)) == e

result = 0
for i in range(1,30):
    for j in range (1,10):
        if check(j,i):
            result += 1

print(result)